### ISSUE TITLE
Multiple versions of package installed, pnpm claims only one version

### ISSUE THREAD

**Issue by runspired** [pnpm/pnpm#5387]
I have a monorepo project using workspaces, multiple of which depends on `@types/ember`.

All dependencies and all workspaces specify precisely the same version of `@types/ember` (4.0.1)

`@types/ember` has a child dependency of `@types/ember__engine` at version `*` due to how DefinitelyTyped specifies dependencies. Latest version of this is `4.0.2`.

During install, the root `node_modules/@types/ember__engine` package receives `4.0.0`, all other locations, for instance `node_modules/.pnpm/@types+ember__application@4.0.1_@babel+core@7.19.0/node_modules/@types/ember__engine` receive latest 4.0.2

`pnpm list @types/ember__engine -r --depth=1000` and `pnpm why @types/ember__engine` both claim only 4.0.2 is present. The lockfile similarly claims only 4.0.2 is present. `tsc` knows otherwise.

**Comment by runspired** [comment]
A further note, using resolutions/overrides does not fix this :(

**Comment by runspired** [comment]
The fix here was to totally blow away the lock file and node_modules and regenerate.

**Comment by runspired** [comment]
I suspect the underlying root issue here was that when the pnpm-lock was originally generated we were migrating from a yarn lock file and had node_modules installed. It appears in this situation that the lock file for pnpm includes hoisted versions of deps it would not include had the yarn lockfile and existing node_modules folder not been present.

**Comment by Kolobok12309** [comment]
@runspired You found any solution?

**Comment by runspired** [comment]
No, I periodically hit this in some re-install situations too where pnpm/lockfile say one thing and node_modules says another. There's some verify check somewhere that has a bug I'm just not sure what.

**Comment by Kolobok12309** [comment]
In our case we forgot add one of packets to `package.json` `dependencies` and use it from deps of another dependency. After adding packet to `dependencies`(or `devDependencies` i think), all work fine, and `node_modules` contain that version

**Comment by zkochan** [comment]
Closing for lack of a reproduction on a supported version.

Reported against pnpm 7.12 with `*` ranges pulling a second copy. Verified while checking 4088 that a `*` subdependency now reuses the version already pinned; a current case needs its own reproduction.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
